### ISSUE TITLE
RFP: MessageAttribute - MinMessageGap

### ISSUE THREAD

**Issue by qdot** [buttplugio/buttplug#106]
**Message Fields:**

For DeviceList/DeviceAdded MessageAttributes:
- MinMessageGap (u32): Minimum time in milliseconds between messages. 0 for no minimum. 

**Describe message usage scenario**

We currently have problems with applications saturating devices. For instance, ScriptPlayer and Syncydink can send vibration commands faster than the server can handle, meaning we buffer vibraiton commands and they play back with lag. This message attribute would allow us to tell the client how often a message can be sent and possibly implement timers on the client side to stop sending in that gap. Similarly, we can implement this in the proprietary protocol on the server to make sure we don't flood whatever communication bus we're on.

**Comment by blackspherefollower** [comment]
Can we really define this per device? I thought the issue was more about the overall ble throughput.

Would it make more sense to record the time the message was received and try and track drift between receipt and execution? That assumes we have a threaded producer/consumer model of course, which I'm not sure we do today.

**Comment by qdot** [comment]
Yeah, this may be a bit much. I was considering it because we have such huge gaps with, say, Lovense (~10-15hz update) versus the Joycon (~63hz update), but I'm not sure this would be the right approach to bridge across both desktop and mobile. For some devices (like Lovense) we could probably track time between sending and status notification, which we don't right now.
